workflow/scripts/osemosys_global/powerplant/user_defined_capacity.py

```diff
@@ -0,0 +1,255 @@
+                                if not df_res_cap_ud_final.empty else None])
+                
+        # For technologies with start year at or after model start year, add to 
+        # TotalAnnualMinCapacityInvestment      
+        df_min_cap_inv = df_min_cap_inv.loc[df_min_cap_inv['YEAR'] >= min(get_years(start_year, end_year))]
+        df_min_cap_inv.drop_duplicates(subset=['REGION', 
+                                               'TECHNOLOGY',
+                                               'YEAR'],
+                                       keep='last',
+                                       inplace=True)
+        
+        tech_list = list(tech_capacity_df['TECHNOLOGY'].unique())
+       
+        df_iar_custom = pd.DataFrame(list(itertools.product(tech_list,
+                                                        [1, 2],
+                                                        get_years(start_year, end_year))
+                                      ),
+                                 columns = ['TECHNOLOGY',
+                                            'MODE_OF_OPERATION',
+                                            'YEAR']
+                                 )
+        df_oar_custom = pd.DataFrame(list(itertools.product(tech_list,
+                                                        [1, 2],
+                                                        get_years(start_year, end_year))
+                                      ),
+                                 columns = ['TECHNOLOGY',
+                                            'MODE_OF_OPERATION',
+                                            'YEAR']
+                                 )  
+        # IAR in modes 1 and 2 are primary electricity commodity ('ELC*01') in 
+        # node_from and node_to, respectively. 
+        # OAR is the inverse of the above
+        df_iar_custom.loc[df_iar_custom['MODE_OF_OPERATION']==1,
+                                        'FUEL'] = ('ELC' + 
+                                                   df_iar_custom['TECHNOLOGY'].str[3:8] + 
+                                                   '01')
+        df_iar_custom.loc[df_iar_custom['MODE_OF_OPERATION']==2,
+                                        'FUEL'] = ('ELC' + 
+                                                   df_iar_custom['TECHNOLOGY'].str[8:13] + 
+                                                   '01')
+        df_oar_custom.loc[df_iar_custom['MODE_OF_OPERATION']==1,
+                                        'FUEL'] = ('ELC' + 
+                                                   df_oar_custom['TECHNOLOGY'].str[8:13] + 
+                                                   '01')
+        df_oar_custom.loc[df_iar_custom['MODE_OF_OPERATION']==2,
+                                        'FUEL'] = ('ELC' + 
+                                                   df_oar_custom['TECHNOLOGY'].str[3:8] + 
+                                                   '01')
```

Comment:
Is this true? Like renewable plants only operate on one mode of operation? Or do we filter that out later on (in which case, thats totally fine)

Reply:
I think we need someway to flag this is only functional for thermal powerplants

Reply:
@trevorb1 I think you changed these lines in one of your commits. Is this still an open question?